### ISSUE TITLE
Should dominator tree be borrowed mutable?

### ISSUE THREAD

**Issue by antonromanov1** [bytecodealliance/wasmtime#5360]
https://github.com/bytecodealliance/wasmtime/blob/37c3c5b1e0ca0c9cece48881d71647c1189979f6/cranelift/codegen/src/dce.rs#L14
Each call inside (`is_valid`, `cfg_postorder`) is **immutable**.

**Comment by antonromanov1** [comment]
@cfallin please take a look.

**Comment by cfallin** [comment]
Yep, it looks like we could modify the function to take an immutable borrow instead. Happy to review a PR if you want to make one.
